### PR TITLE
Remove query params from target filename on getFile (fixes #3957)

### LINKED ISSUE #3957: http.getFile(url) should strip query string from saved file name
I'm trying to download files that have some long query string parameters and I'm using `http.getFile` w/o specifying a local file name. The default file name that is picked by the framework includes both the file name and the query string and thus the save fails. 

Expected:
The default file name that is generated should strip out any query string from the URL and just save with a file name that matches the url.

### PR DESCRIPTION
**Pull request by vjoao**
### The commit message references a specific issue in this repo
Fixes #3957 

Updated tests to better reflect the getFile with a query string;

### CODE REVIEW

**Comment by vjoao** [comment]
run ci

File: tns-core-modules/http/http-request/http-request.android.ts [hunk 1]
```diff
@@ -123,7 +123,17 @@ function onRequestComplete(requestId: number, result: org.nativescript.widgets.A
                 var fs: typeof fsModule = require("file-system");
                 var fileName = callbacks.url;
                 if (!destinationFilePath) {
-                    destinationFilePath = fs.path.join(fs.knownFolders.documents().path, fileName.substring(fileName.lastIndexOf('/') + 1));
+                    let slashPos = fileName.lastIndexOf('/') + 1;
```
Comment (vakrilov) [inline tns-core-modules/http/http-request/http-request.android.ts]: Can you move the common logic directly in the `http.getFile()` method

File: tns-core-modules/http/http-request/http-request.ios.ts [hunk 2]
```diff
@@ -152,7 +152,16 @@ export function request(options: http.HttpRequestOptions): Promise<http.HttpResp
                                     var fs: typeof fsModule = require("file-system");
                                     var fileName = options.url;
                                     if (!destinationFilePath) {
-                                        destinationFilePath = fs.path.join(fs.knownFolders.documents().path, fileName.substring(fileName.lastIndexOf('/') + 1));
+                                        let slashPos = fileName.lastIndexOf('/') + 1;
```
Comment (vakrilov) [inline tns-core-modules/http/http-request/http-request.ios.ts]: Can you move the common logic directly in the http.getFile() method

**Comment by vakrilov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by vakrilov** [comment]
run ci

**Comment by vakrilov** [comment]
run ci

**Comment by vakrilov** [comment]
run ci

**Comment by vjoao** [comment]
@vakrilov, can you review the URLs of the http tests? They are getting timeouts and the tests are failing

**Comment by vakrilov** [comment]
run ci

**Comment by vakrilov** [comment]
@vjoao Sure - I'm still not sure what the problem is. Will ping you if there is anything you can do.

**Comment by vchimev** [comment]
Hey @vjoao,

could you please rebase your branch on **master** of https://github.com/NativeScript/NativeScript? This would help the Travis CI build (Android) to pass. 

However, for iOS the unit tests app crashes with `Segmentation falut: 11`:
```
CONSOLE INFO file:///app/tns_modules/tns-core-modules/trace/trace.js:159:29: Test: --- [HTTP.test_request_isDefined] OK, duration: 0.13
CONSOLE INFO file:///app/tns_modules/tns-core-modules/trace/trace.js:159:29: Test: --- [HTTP.test_request_shouldFailIfOptionsUrlIsNotDefined] OK, duration: 11.75297999382019
CONSOLE INFO file:///app/tns_modules/tns-core-modules/trace/trace.js:159:29: Test: --- [HTTP.test_request_requestShouldTimeout] OK, duration: 504.55438500642776
CONSOLE INFO file:///app/tns_modules/tns-core-modules/trace/trace.js:159:29: Test: --- [HTTP.test_request_responseStatusCodeShouldBeDefined] OK, duration: 151.29155802726746
CONSOLE INFO file:///app/tns_modules/tns-core-modules/trace/trace.js:159:29: Test: --- [HTTP.test_request_responseHeadersShouldBeDefined] OK, duration: 144.48429000377655
CONSOLE INFO file:///app/tns_modules/tns-core-modules/trace/trace.js:159:29: Test: --- [HTTP.test_request_responseContentShouldBeDefined] OK, duration: 142.1056840121746
CONSOLE INFO file:///app/tns_modules/tns-core-modules/trace/trace.js:159:29: Test: --- [HTTP.test_request_responseContentToStringShouldReturnString] OK, duration: 139.66580998897552
CONSOLE INFO file:///app/tns_modules/tns-core-modules/trace/trace.js:159:29: Test: --- [HTTP.test_request_responseContentToJSONShouldReturnJSON] OK, duration: 141.81639802455902
CONSOLE INFO file:///app/tns_modules/tns-core-modules/trace/trace.js:159:29: Test: --- [HTTP.test_request_responseContentToImageShouldReturnCorrectImage] OK, duration: 141.09768798947334
May 26 13:44:04 mcsofvchimev com.apple.CoreSimulator.SimDevice.A7E6EB91-8A22-48C9-926A-DC31444CFC0A.launchd_sim[34410] (UIKitApplication:org.nativescript.tests[0xcfe0][35925]): Service exited due to Segmentation fault: 11
```

**Comment by vakrilov** [comment]
Hey @vjoao, I just opened another PR #4291 with your changes with small modifications to assure backwards-compatibility and satisfy the failing tests.

Thanks again for the great contribution!

**Pull request closed without merging**